### PR TITLE
[CBRD-23489] Double write buffer

### PR DESCRIPTION
**Pull request by ribram**
http://jira.cubrid.org/browse/CBRD-23489

I've added a draft of changes. System parameters need some clarifications before I can finalize the description of dwb file.

Questions:

  - are parameters changeable? the are marked as user change, but it seems they are only used on database creation; there is a function that could recreate the DWB buffer/file with a different size and number of blocks, but it is never used.
  - the blocks number can be set to any value between 0 and 32. 0 disables DWB, 1 makes a single block, default is 2. I think 2 was the only tested value and 1 is a dangerous option (can dwb receive more pages while it is flushing to disk in this case?). My opinion is to either hide or set minimum value to two. Please advise.
  - the thread enable/disable is unclear to me how it works. What happens if DWB is created, but thread is disabled?

### CODE REVIEW

**Comment by alexstanro** [comment]
1. recreate exists, currently not used. It may be activated, if we want the behavior. It is safe.
2. 1 block is safe, but may slow down the system (adding page may wait for flush). 
3. Only thread enabled was tested.

**Comment by ribram** [comment]
@alexstanro Thanks for the explanations, how each parameter works is now clear.

The question of exposing block count and DWB thread configurations remains. We did not test all configurations, is it safe for production?

We have next options:

  1. document configurations and warn against changing default values
  2. hide configurations

@eseokoh What do you think?

**Comment by eseokoh** [comment]
I'd like ti propose:
change the param as const int 
or
leave them as undocumented param.

**Comment by ribram** [comment]
@eseokoh 

Can you please explain how to "change the param as const int"?

**Comment by eseokoh** [comment]
I meant remove it from sys params and
make it `const int ???`;

**Comment by ribram** [comment]
> I meant remove it from sys params and
> make it `const int ???`;

Ah, I see. I would prefer to just hide the parameters for now. Maybe we'll want to test different configurations in the future.

**Comment by ribram** [comment]
Regardless of what we decide for parameters, they don't have to to be documented in the manual. Please review updated entries.

File: en/admin/db_manage.rst [hunk 1]
```diff
@@ -106,6 +106,9 @@ The following is an example of files related to the database when *testdb* datab
 +----------------+-------+-----------------+                +------------------------------------------------------------------------------------------------------+
 | testdb_lgat    | 512MB | | active        |                | | Active log file                                                                                    |
 +----------------+-------+-----------------+----------------+------------------------------------------------------------------------------------------------------+
+| testdb_dwb     |  1MB  | | permanent     | | Double write | Double write buffer storage file, where flushed pages are written first.                             |
```
Comment (eseokoh) [inline en/admin/db_manage.rst]: I think it is rather temporary.

Comment (ribram) [reply en/admin/db_manage.rst]: fixed

**Pull request merged**